conan/tools/_compilers.py

```diff
@@ -236,24 +236,32 @@ def _cppstd_apple_clang(clang_version, cppstd):
         v14 = "c++1y"
         vgnu14 = "gnu++1y"
 
-    if Version(clang_version) >= "6.1":
-        v17 = "c++1z"
-        vgnu17 = "gnu++1z"
-
+    # Not confirmed that it didn't work before 9.1 but 1z is still valid, so we are ok
+    # Note: cmake allows c++17 since version 10.0
     if Version(clang_version) >= "9.1":
-        # Not confirmed that it didn't work before 9.1 but 1z is still valid, so we are ok
         v17 = "c++17"
         vgnu17 = "gnu++17"
+    elif Version(clang_version) >= "6.1":
+        v17 = "c++1z"
+        vgnu17 = "gnu++1z"
 
-    if Version(clang_version) >= "10.0":
+    if Version(clang_version) >= "13.0":
+        v20 = "c++20"
+        vgnu20 = "gnu++20"
+    elif Version(clang_version) >= "10.0":
         v20 = "c++2a"
         vgnu20 = "gnu++2a"
 
+    if Version(clang_version) >= "13.0":
+        v23 = "c++2b"
+        vgnu23 = "gnu++2b"
```

Comment:
Why don't you merge the two `if`?

Reply:
I've followed current structure of the code. There's one section per c++ standard version. Those if are the same, but the latter will be converted to if / elif once "c++23" switch is accepted.